src/js/utils/date-utils.js

```diff
@@ -0,0 +1,283 @@
+        var DateUtils, isValid, dateObject, userTimezone, localizeTime, timeDisplay, localize;
+
+        /**
+         * Helper function to determine variable/context validity.
+         *
+         * @param {string} candidateVariable to test
+         * @returns {boolean}
+         */
+        isValid = function(candidateVariable) {
+            var valid = true;
+            if (candidateVariable === undefined) {
+                valid = false;
+            } else if (candidateVariable === 'undefined') {
+                valid = false;
+            } else if (candidateVariable === '') {
+                valid = false;
+            } else if (candidateVariable === 'Invalid date') {
+                valid = false;
+            } else {
+                valid = true;
+            }
+            return valid;
+        };
+
+        /**
+         * Attempt to convert a string to mutable 'moment' datetime object.
+         *
+         * The input datetimeString can be just a date, a date and time,
+         * or a complex date/time string with TZ offset. This is fairly wide-open, as
+         * many strings will satisfy this requirement.
+         * (e.g. 01-01-16 and Jan 1, 2016, and 2016-01-01 01:00:00+00:00 will work)
+         *
+         * The recommendation is to use a fully informed string, e.g. '2016-01-01 01:00:00+00:00',
+         * to limit possible date, time, and timezone ambiguities.
+         *
+         * @param {string} datetimeString. A string representation of a datetime.
+         * @returns {object} A moment.js UTC datetime object.
+         *
+        */
+        dateObject = function(datetimeString) {
+            var utcDateObject;
+            if (isValid(datetimeString)) {
+                utcDateObject = moment(datetimeString).utc();
+            } else {
+                utcDateObject = '';
+            }
+            return utcDateObject;
+        };
+
+        /**
+         * Attempt to determine user's timezone from preferences, then browser settings.
+         * Will default to UTC.
+         *
+         * @param {string} timezone 'Region/Population_Center' e.g. 'America/New_York'.
+         * If this is None or undefined, we attempt to determine the TZ from system/browser settings.
+         * @return {string} Will attempt to return, in this order, the entered, determined, and default (UTC)
+         * timezone.
+         *
+         */
+        userTimezone = function(timezone) {
+            var determinedZone;
+            if (isValid(timezone)) {
+                determinedZone = timezone;
+            } else {
+                determinedZone = momentTZ.tz.guess();
+            }
+
+            if (isValid(determinedZone) === false) {
+                determinedZone = 'UTC';
+            }
+            return determinedZone;
+        };
+
+        /**
+         *
+         * Attempt to offset original time based on timezone.
+         *
+         * @param {Object} dateTimeObject A moment.js datetime object.
```

Comment:
Consider removing all "Object" prefixes from your function and variable names, as they don't add much value.  It is assumed that all objects that are passed around are objects.  Rather, if any are specifically, strings, you can include "string" in the name - and have the default assumption be "Object".